backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/configuration/JamesExecutionProfiles.java

```diff
@@ -27,6 +27,17 @@
 import com.datastax.oss.driver.api.core.config.DriverExecutionProfile;
 
 public interface JamesExecutionProfiles {
+    /**
+     * James application configuration allows for some operation to choose
+     * or not to enable LWTs.
+     *
+     * When LWTs are enable underlying DAOs will use LWT execution profiles.
+     */
+    enum ConsistencyChoice {
```

Comment:
How to use this?

```
class MyDAO {

   private final DriverExecutionProfile lwtProfile;
   
   MyDAO(CqlSession session) {
     // ... 
     lwtProfile = JamesExecutionProfiles.getLwtExecutionProfile(session);
   }

  //...
  
  public RESULT operation(Id id, ConsistencyChoice consistencyChoice) {
    BoundStatement boundStatement = statement.bind().setString(ID, id.serialize());
    if (consistencyChoice == STRONG) {
        boundStatement.setExecutionProfile(lwtProfile);
    }
    return executor.execute(boundStatement);
  }

```

Reply:
`setExecutionProfile` doc: 
```
All the driver's built-in implementations are immutable, and return a new instance from this method. However custom implementations may choose to be mutable and return the same instance.
```

I propose: 
```
private BoundStatement setExecutionProfileIfNeeded(BoundStatement statement, JamesExecutionProfiles.ConsistencyChoice consistencyChoice) {
        if (consistencyChoice.equals(STRONG)) {
            return statement.setExecutionProfile(lwtProfile);
        } else {
            return statement;
        }
    }
```

Reply:
@quantranhong1999  what is the difference?
BoundStatement is unique for each query?